src/main/webapp/app/exercises/shared/structured-grading-criterion/grading-instructions-details/grading-instructions-details.component.ts

```diff
@@ -285,4 +320,113 @@ export class GradingInstructionsDetailsComponent implements OnInit {
         this.exercise.gradingCriteria = [];
         this.createSubInstructionCommands(domainCommands);
     }
+
+    /**
+     * @function onInstructionChange
+     * @desc 1. Gets a tuple of text and domainCommandIdentifiers and assigns text values according to the domainCommandIdentifiers
+     *       2. The tupple order is the same as the order of the commands in the markdown text inserted by the user
+     * @param domainCommands containing tuples of [text, domainCommandIdentifiers]
+     * @param {GradingInstruction} instruction
+     * @param {GradingCriterion} criteria
+     */
+    onInstructionChange(domainCommands: [string, DomainCommand | null][], instruction: GradingInstruction): void {
+        this.instructions = [];
+        this.instructions.push(instruction);
```

Comment:
Why not just put inside it?

Reply:
want to be sure following `setInstructionParameters` runs for only this instruction. so, i just initialized with an empty array every time `onInstructionChange` is called